lib/valueflow.cpp

```diff
@@ -2253,10 +2253,14 @@ struct ValueFlowAnalyzer : Analyzer {
         } else if (getSettings()->library.getFunction(tok)) {
             // Assume library function doesn't modify user-global variables
             return Action::None;
-            // Function cast does not modify global variables
+        } else if (Token::simpleMatch(tok->astParent(), ".") && tok->astParent()->astOperand1() && tok->astParent()->astOperand1()->variable() &&
```

Comment:
We shouldn't do `tok->astParent()->astOperand1()->variable()` as containers can be arbitrary expressions. Can you just do `astIsContainer(tok->astParent()->astOperand1())`?